test/helpers/chaiWeb3.js

```diff
@@ -0,0 +1,26 @@
+export default function(chai, utils) {
+  chai.Assertion.addMethod("blockchainArrayOfSize", async function(size) {
+    // I would love to hear ideas for better implementation
+
+    const web3ArrayAccessor = this._obj;
+
+    try {
+      await web3ArrayAccessor(size - 1);
+    } catch (e) {
+      this.assert(
+        false,
+        `expected web3 array to be size of ${size} but it looks like it's smaller` // i think it's impossible to get an array name in this point
```

Comment:
> impossible

```
    const req = web3ArrayAccessor.request(0);
    const bytes4 = req.params[0].data.slice(10);
```

The bytes4 contains a hash of the array name, you can now iterate through all contracts and find the matching one :D